comp/ndmtmp/sender/component.go

```diff
@@ -24,3 +29,28 @@ type Component interface {
 var Module = fxutil.Component(
 	fx.Provide(getDefaultSender),
 )
+
+// MockComponent is an interface satisfied by mocksender.MockSender.
+type MockComponent interface {
+	Component
+	AssertServiceCheck(t *testing.T, checkName string, status servicecheck.ServiceCheckStatus, hostname string, tags []string, message string) bool
+	AssertMetric(t *testing.T, method string, metric string, value float64, hostname string, tags []string) bool
+	AssertMonotonicCount(t *testing.T, method string, metric string, value float64, hostname string, tags []string, flushFirstValue bool) bool
+	AssertHistogramBucket(t *testing.T, method string, metric string, value int64, lowerBound float64, upperBound float64, monotonic bool, hostname string, tags []string, flushFirstValue bool) bool
+	AssertMetricInRange(t *testing.T, method string, metric string, min float64, max float64, hostname string, tags []string) bool
+	AssertMetricTaggedWith(t *testing.T, method string, metric string, tags []string) bool
+	AssertMetricNotTaggedWith(t *testing.T, method string, metric string, tags []string) bool
+	AssertEvent(t *testing.T, expectedEvent event.Event, allowedDelta time.Duration) bool
+	AssertEventPlatformEvent(t *testing.T, expectedRawEvent []byte, expectedEventType string) bool
+	AssertEventMissing(t *testing.T, expectedEvent event.Event, allowedDelta time.Duration) bool
+}
+
+// MockModule provides a MockSender as the sender Component.
+var MockModule = fxutil.Component(
+	fx.Provide(
+		newMockSender,
+		func(s MockComponent) Component {
+			return s
+		},
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Is that a way to "cast" a MockComponent to Component ?
So, in the end, the same object is used/injected as both MockComponent and Component ?

Reply:
Yes - if `newMockSender` returned `Component` instead of `MockComponent`, then the tests would have to retrieve the component and cast it to a mockcomponent in order to use any of the mock-specific methods. But since it returns `MockComponent`, fx won't see anything providing `Component`. So this function provides `Component` by converting the mock component to a component.

Although as I typed this response, I just realized I don't need to do this if I just have `newMockSender` return `(MockComponent, Component)`, in which case it will provide both at once, so I'll change it to do that.